src/main/java/seedu/address/model/schedule/EventCollideWithTimeDatePredicate.java

```diff
@@ -0,0 +1,53 @@
+package seedu.address.model.schedule;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import seedu.address.model.person.Person;
+
+/**
+ * Tests that a {@code Person}'s {@code Schedule} does not collide to given time and/or date.
+ */
+public class EventCollideWithTimeDatePredicate implements Predicate<Person> {
```

Comment:
Is it more intuitive to name the class IsPersonBusy?

Reply:
Also, it was a pretty good call to use Java's in built Predicate class imo.

Reply:
This predicate actually checks if the person is free